macros/blueprint-proc-macro/src/lib.rs

```diff
@@ -81,7 +98,14 @@ pub fn registration_hook(args: TokenStream, input: TokenStream) -> TokenStream {
     }
 }
 
-/// TDOD: Documentation
+/// A procedural macro that annotates a function as a request hook, mainly used
+/// for the metadata in the `blueprint.json`.
+/// # Example
+/// ```rust,no_run
+/// # use gadget_blueprint_proc_macro::request_hook;
+/// #[request_hook(evm = "MyRequestHook")]
+/// pub fn my_request_hook(nft_id: u32);
```

Comment:
```suggestion
/// pub fn my_request_hook();
```

Same here, please avoid superfluous details. Focus and clarity are really important here to ensure we're on the same page.